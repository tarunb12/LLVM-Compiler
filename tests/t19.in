// Pythagorean theorem

int power(int x, int n) {
    if (n == 0) return 1
    return x * power(x, n - 1)
}

int flooredSqrt(int x) {
    if (x == 0 || x == 1) return x
    int i = 1
    int result = 1
    while (result <= x) {
        i++
        result = i * i
    }
    return i - 1
}

int main() {
    int a = 5
    int b = 12
    int c = flooredSqrt(power(a, 2) + power(b, 2))
    printf("Right triangle sides: %d %d %d", a, b, c)
}