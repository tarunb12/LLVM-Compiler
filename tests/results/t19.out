Error: File "tests/t19.in": No entry point found in the file. Define the entry point with the function main defined as follows:

int main() {
	...
	return 0
}
.